File: Big-DicTaphone/Models/Recording.swift
import Foundation

/// Represents a single voice recording with its metadata and processing status
struct Recording: Identifiable, Codable {
    let id: UUID
    var title: String
    let date: Date
    var duration: TimeInterval
    let audioFileName: String
    let language: RecordingLanguage
    var transcription: String?
    var summary: MeetingSummary?
    var status: ProcessingStatus
    
    init(
        id: UUID = UUID(),
        title: String = "",
        date: Date = Date(),
        duration: TimeInterval = 0,
        audioFileName: String,
        language: RecordingLanguage = .auto,
        transcription: String? = nil,
        summary: MeetingSummary? = nil,
        status: ProcessingStatus = .recorded
    ) {
        self.id = id
        self.title = title.isEmpty ? Self.defaultTitle(for: date) : title
        self.date = date
        self.duration = duration
        self.audioFileName = audioFileName
        self.language = language
        self.transcription = transcription
        self.summary = summary
        self.status = status
    }
    
    /// Generate a default title based on the recording date
    static func defaultTitle(for date: Date) -> String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .short
        return "Recording \(formatter.string(from: date))"
    }
    
    /// Get the full URL for the audio file
    var audioURL: URL {
        Recording.recordingsDirectory.appendingPathComponent(audioFileName)
    }
    
    /// Directory where all recordings are stored
    static var recordingsDirectory: URL {
        let documentsPath = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
        let recordingsPath = documentsPath.appendingPathComponent("Recordings")
        
        // Create directory if it doesn't exist
        if !FileManager.default.fileExists(atPath: recordingsPath.path) {
            try? FileManager.default.createDirectory(at: recordingsPath, withIntermediateDirectories: true)
        }
        
        return recordingsPath
    }
    
    /// Formatted duration string (MM:SS or HH:MM:SS)
    var formattedDuration: String {
        let hours = Int(duration) / 3600
        let minutes = (Int(duration) % 3600) / 60
        let seconds = Int(duration) % 60
        
        if hours > 0 {
            return String(format: "%d:%02d:%02d", hours, minutes, seconds)
        } else {
            return String(format: "%d:%02d", minutes, seconds)
        }
    }
}

/// Supported languages for transcription
enum RecordingLanguage: String, Codable, CaseIterable, Identifiable {
    case auto = "auto"
    case dutch = "nl-NL"
    case english = "en-US"
    case englishUK = "en-GB"
    
    var id: String { rawValue }
    
    var displayName: String {
        switch self {
        case .auto: return "Auto-detect"
        case .dutch: return "Nederlands"
        case .english: return "English (US)"
        case .englishUK: return "English (UK)"
        }
    }
    
    var flagEmoji: String {
        switch self {
        case .auto: return "🌐"
        case .dutch: return "🇳🇱"
        case .english: return "🇺🇸"
        case .englishUK: return "🇬🇧"
        }
    }
}

/// Processing status of a recording
enum ProcessingStatus: String, Codable {
    case recorded = "Recorded"
    case transcribing = "Transcribing..."
    case summarizing = "Summarizing..."
    case complete = "Complete"
    case failed = "Failed"
    case mediaMissing = "Audio Missing"
    
    var color: String {
        switch self {
        case .recorded: return "gray"
        case .transcribing, .summarizing: return "orange"
        case .complete: return "green"
        case .failed, .mediaMissing: return "red"
        }
    }
    
    var isProcessing: Bool {
        self == .transcribing || self == .summarizing
    }
}

File: android/app/src/main/java/com/bigdictaphone/app/data/Recording.kt
package com.bigdictaphone.app.data

import kotlinx.serialization.Serializable
import java.util.UUID

/**
 * Represents a single voice recording with its metadata and processing status
 */
@Serializable
data class Recording(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val timestamp: Long = System.currentTimeMillis(),
    val duration: Long = 0, // Duration in milliseconds
    val audioFileName: String,
    val language: RecordingLanguage = RecordingLanguage.AUTO,
    val transcription: String? = null,
    val summary: MeetingSummary? = null,
    val status: ProcessingStatus = ProcessingStatus.RECORDED,
    val transcriptionMode: TranscriptionMode? = null,
    val processingError: String? = null,
    val jobStatus: JobStatus = JobStatus.NONE,
    val deliveryStatus: DeliveryStatus = DeliveryStatus.NONE,
    val captureStatus: CaptureStatus = CaptureStatus.SAVED
) {
    val formattedDuration: String
        get() {
            val totalSeconds = duration / 1000
            val hours = totalSeconds / 3600
            val minutes = (totalSeconds % 3600) / 60
            val seconds = totalSeconds % 60
            
            return if (hours > 0) {
                String.format(java.util.Locale.getDefault(), "%d:%02d:%02d", hours, minutes, seconds)
            } else {
                String.format(java.util.Locale.getDefault(), "%d:%02d", minutes, seconds)
            }
        }
}

/**
 * Supported languages for transcription
 */
@Serializable
enum class RecordingLanguage(val locale: String, val displayName: String, val flag: String) {
    AUTO("auto", "Auto-detect", "🌐"),
    DUTCH("nl-NL", "Nederlands", "🇳🇱"),
    ENGLISH("en-US", "English (US)", "🇺🇸"),
    ENGLISH_UK("en-GB", "English (UK)", "🇬🇧")
}

/**
 * Processing status of a recording
 */
@Serializable
enum class ProcessingStatus(val displayName: String) {
    RECORDED("Recorded"),
    TRANSCRIBING("Transcribing..."),
    SUMMARIZING("Summarizing..."),
    COMPLETE("Complete"),
    FAILED("Failed");
    
    val isProcessing: Boolean
        get() = this == TRANSCRIBING || this == SUMMARIZING
}

@Serializable
enum class JobStatus { NONE, QUEUED, RUNNING, RETRYABLE, CANCELLED, COMPLETE }

@Serializable
enum class DeliveryStatus { NONE, SENDING, SENT, FAILED, UNCERTAIN }

@Serializable
enum class CaptureStatus { ACTIVE, PAUSED, PENDING, SAVED, INTERRUPTED }
